runtime/delegates_test.go

```diff
@@ -23,106 +22,102 @@ func TestDelegateRuntimeAPIs(t *testing.T) {
 		t.Parallel()
 
 		env, err := testutils.Setup()
-		if err != nil {
-			t.Skipf("Failed to setup test environment: %v", err)
-		}
+		require.NoError(t, err, "Failed to setup test environment")
 		defer env.Teardown()
 
 		blockHash, err := env.Client.Api.RPC.Chain.GetBlockHashLatest()
 		require.NoError(t, err, "Failed to get latest block hash")
 
 		delegates, err := runtime.GetDelegates(env.Client, &blockHash)
+		require.NoError(t, err, "Failed to get delegates")
 
-		if err == nil && len(delegates) > 0 {
+		switch len(delegates) {
+		case 0:
+			t.Log("No delegates found in test environment, testing with random account")
+		default:
 			delegateAccount := delegates[0].AccountID
 
-			delegate, err := runtime.GetDelegate(env.Client, delegateAccount, &blockHash)
+			delegateOption, err := runtime.GetDelegate(env.Client, delegateAccount, &blockHash)
 			require.NoError(t, err, "Failed to get specific delegate")
-			assert.NotNil(t, delegate, "Delegate should not be nil")
+			assert.NotNil(t, delegateOption, "Delegate option should not be nil")
+
+			ok, delegate := delegateOption.Unwrap()
+			require.True(t, ok, "Delegate should exist")
 
 			assert.Equal(t, delegateAccount, delegate.AccountID, "Delegate accounts should match")
 			assert.Equal(t, delegates[0].TakeRate, delegate.TakeRate, "TakeRate values should match")
 			assert.Equal(t, delegates[0].TotalStake, delegate.TotalStake, "TotalStake values should match")
 
 			t.Logf("Successfully retrieved delegate: %x", delegate.AccountID.ToBytes())
-		} else {
-			t.Log("No delegates found in test environment to test GetDelegate with valid account")
+
+			delegateOption, err = runtime.GetDelegate(env.Client, delegates[0].AccountID, nil)
+			require.NoError(t, err, "Should work with nil block hash")
+			assert.NotNil(t, delegateOption, "Delegate option should not be nil")
 		}
 
 		randomKeyring, err := signature.KeyringPairFromSecret("//Random/Test/Account/12345", 42)
 		require.NoError(t, err)
 		randomAccount, err := types.NewAccountID(randomKeyring.PublicKey)
 		require.NoError(t, err)
 
-		delegate, err := runtime.GetDelegate(env.Client, *randomAccount, &blockHash)
-		assert.Error(t, err, "Should error on non-existent delegate")
-		assert.Nil(t, delegate, "Delegate should be nil on error")
-		errorMsg := err.Error()
-		isExpectedError := strings.Contains(errorMsg, "no delegate found") ||
-			strings.Contains(errorMsg, "Invalid params")
-		assert.True(t, isExpectedError, "Error should indicate delegate not found or invalid params")
-
-		if len(delegates) > 0 {
-			delegate, err = runtime.GetDelegate(env.Client, delegates[0].AccountID, nil)
-			assert.Error(t, err, "Should error with nil block hash")
-			assert.Contains(t, err.Error(), "block hash cannot be nil", "Should have proper error message for nil block hash")
-			assert.Nil(t, delegate, "Delegate should be nil on error")
+		delegateOption, err := runtime.GetDelegate(env.Client, *randomAccount, &blockHash)
+		switch {
```

Comment:
Why are you using a switch here? Just assert the error is nil like you have done before.